Activities/GAScripted.cpp

```diff
@@ -204,6 +204,42 @@ int GAScripted::ReloadScripts() {
     return error;
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool GAScripted::ActivityCanBeSaved() const {
```

Comment:
This is a big and ugly and messy function. May I suggest that we do this lazily instead - try calling OnSave() and just check if it fails? It'd put some initial values in the SavedNumbers / SavedStrings map, but these should be overwritten on the next save regardless.

Reply:
Yeah, it's pretty ugly haha. I basically just copied the SceneIsCompatible check, but cut out the stuff I didn't need.
I actually tried a oneline check (using luaman.globalexists) but wasn't getting it to work, and I didn't wanna keep bothering when I had something that works.

Your idea might work too, though I think it's kinda a pain to pass the result of whether the function exists from lua to cpp. I can give it another look though I guess.

Reply:
It's not a huge deal. The only other issue is that this doesn't handle Lua's multiline comments properly (they can be nested in some cases), but that's extremely minor and nobody uses those nested anyways.

Reply:
I'll chuck in a todo at least, and maybe take another look at it